### ISSUE TITLE
New Galaxy tool parameters

### ISSUE THREAD

**Issue by bgruening** [galaxyproject/galaxy#6806]
I will use this issue to collect some ideas of useful new Galaxy tool parameters. Please add new ideas as well.

* date picker (useful for ecology and meteorology, https://github.com/charliekassel/vuejs-datepicker)
 xref https://github.com/galaxyproject/galaxy/issues/6223
* location picker (lat/long (+altitude?) or graphical: https://www.npmjs.com/package/vue-location-picker)
* range picker, e.g. 1:5,6,7, this is currently a simple text field with a sanitizer, I'm wondering if this is so common that we should have an own param.
   ```
      <sanitizer invalid_char="">
          <valid initial="string.digits">
             <add value="_" />
             <add value="," />
          </valid>
      </sanitizer>
   ```

**Comment by bernt-matthias** [comment]
:+1: for the range picker, in particular version having a min max field could be handy for some tools.

**Comment by bernt-matthias** [comment]
I often end up creating a conditional to create a data param for single files, collections and paired collections. Would be nice to have a shortcut for this. Maybe along something along the lines of 
```<param type="data,pair" />```

**Comment by bgruening** [comment]
Just had a discussion about the date-picker. For a few people, it would be also very useful to not only pick days, weeks and so on, but also to select 2 million years ago until 1 million years ago. I'm not sure how feasible this is for standard date-pickers, but maybe we can consider this use-case as well.

**Comment by bgruening** [comment]
xref: https://github.com/galaxyproject/galaxy/commit/d43b1ed0c21cbcbb3228409b7d768968b2d0f69f (implements the color-picker)

**Comment by annefou** [comment]
Yes it would be very useful. I don't know much about javascript, so the links below may not be very relevant...
- https://scholarslab.lib.virginia.edu/blog/parsing-bc-dates-with-javascript/
- https://www.ama3.com/anytime/

**Comment by bernt-matthias** [comment]
Would be nice to have a kind of select where the user could add further entries. That is, the options are suggestions (or defaults).

In this context also typed selects could be nice. That is where only float or integer values are possible. 

Maybe the already present possibility to add option tags to text inputs could be extended by allowing this also for integer and float parameters + allowing multiple true.

**Comment by jennaj** [comment]
A ticket I made about how a dataset picker could work... just some ideas, not stuck on how actually implemented :)
https://github.com/galaxyproject/galaxy/issues/7436

**Comment by bernt-matthias** [comment]
Adding to my previous suggestion for a select where users can add entries: Since the generation of dynamic options only uses the first 1MB for extracting the options this should actually be the default for such options, because the extraction might miss necessary cases.

**Comment by bernt-matthias** [comment]
Alternatively, the possibility to specify the (distinct) default values of multiple repeat elements would be nice (for repeats with `default` > 1). Currently if a repeat element has a `value` attribute this is used as the default for each new repeat element.

**Comment by FredericBGA** [comment]
Hi,

it might be nice to be able to set a min/max value on a single sliding bar.
Like here: [plantgenomeediting.net](http://www.plantgenomeediting.net/)

**Comment by astrovsky01** [comment]
Also, a multiple-true conditional would be nice. There are many parameters in tools that have shared options for some but not all of the values.
For example,
```
<when value="asdf" || value="qwer">
    <param.../>
</when>
```

**Comment by bernt-matthias** [comment]
> Also, a multiple-true conditional would be nice. There are many parameters in tools that have shared options for some but not all of the values. For example,
> 
> ```
> <when value="asdf" || value="qwer">
>     <param.../>
> </when>
> ```

Indeed. See https://github.com/galaxyproject/galaxy/issues/7173 :) Lets continue over there

I guess the best would be (as agreed in the 2022 Montpellier meeting) to open separate issues and work out the tool xml syntax end tool semantics.
